.github/workflows/update-bff-dependency/src/repository-helper.js

```diff
@@ -0,0 +1,305 @@
+      });
+      core.debug(`Blob created with sha ${blob.sha}`);
+      return blob;
+    } catch (error) {
+      throw new Error(`Error during blob creation: ${error}`);
+    }
+  }
+
+  async #createBranchTree(branchSha, files) {
+    core.debug(`Creating branch tree`);
+    const shaFiles = [];
+    for (const file of files) {
+      const blob = await this.#createBlob(file.content);
+      shaFiles.push({
+        path: file.path,
+        sha: blob.sha,
+      });
+    }
+    try {
+      const { data: branchTree } = await this.#octokit.rest.git.createTree({
+        owner: github.context.repo.owner,
+        repo: github.context.repo.repo,
+        tree: shaFiles.map((file) => ({
+          path: file.path,
+          mode: "100644",
+          type: "blob",
+          sha: file.sha,
+        })),
+        base_tree: branchSha,
+      });
+      core.debug(`Branch tree created`);
+      return branchTree;
+    } catch (error) {
+      throw new Error(`Error during branch tree creation: ${error}`);
+    }
+  }
+
+  async #updateBranchRef(branchName, commitSha) {
+    core.debug(`Updating branch ${branchName} reference`);
+    try {
+      await this.#octokit.rest.git.updateRef({
+        owner: github.context.repo.owner,
+        repo: github.context.repo.repo,
+        ref: `heads/${branchName}`,
+        sha: commitSha,
+      });
+      core.debug(`Branch ${branchName} reference updated`);
+    } catch (error) {
+      throw new Error(
+        `Error during branch ${branchName} reference updating: ${error}`
+      );
+    }
+  }
+
+  async commitChanges(branchName, files) {
+    if (files.length === 0) {
+      core.info(`Nothing to commit on branch ${branchName}`);
+      return;
+    }
+    core.info(`Committing changes on branch ${branchName}`);
+    // get branch tree
+    const branchTree = await this.#createBranchTree(this.#branchSha, files);
+    core.debug(`Branch tree sha ${branchTree.sha}`);
+    let commit;
+    try {
+      const response = await this.#octokit.rest.git.createCommit({
+        owner: github.context.repo.owner,
+        repo: github.context.repo.repo,
+        message: "Update pn-bff dependency",
+        tree: branchTree.sha,
+        parents: [this.#branchSha],
+      });
+      core.info(`Changes on branch ${branchName} committed`);
+      commit = response.data;
+    } catch (error) {
+      throw new Error(`Error during commit creation: ${error}`);
+    }
+    core.info(`Pushing changes on branch ${branchName}`);
+    await this.#updateBranchRef(branchName, commit.sha);
+    core.info(`Changes on branch ${branchName} pushed`);
+  }
+
+  #getPullRequestBodyString(changesToCommit) {
+    let body =
+      "## Short description\n\nUpdated pn-bff dependencies\n\n## List of changes proposed in this pull request\n\n";
+    for (const change of changesToCommit) {
+      body += `- Updated ${change.path}\n`;
+    }
+    body += "\n## How to test\n\nCheck that tests runs correctly";
```

Comment:
Tests alone don't rerun generate command. Add that, before testing, we have to run the generate command